brave/brave-core
# Hide Brave Rewards button

- Adds a new setting "Hide Brave Rewards button" under the Appearance
  section of settings.

- Modifies BraveActionsContainer to allow hiding the Brave Rewards
  action when the setting is turned on, but only when Brave Rewards is
  disabled. Hides the action for Guest sessions.

- Adds a browser test to verify that the button is hidden when expected.

- Fixes various lint complaints in the affected files.

Fixes brave/brave-browser#2975

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:

**Automated testing**
`npm run test -- brave_browser_tests --filter=BraveActionsContainerTest.HideBraveRewardsAction`
`npm run test -- brave_browser_tests --filter=BraveActionsContainerTest.BraveRewadsActionHiddenInGuestSession`

**Manual testing**
1. Test new settings toggle
- Start Brave browser with a fresh profile and verify that the Brave Rewards action button is shown in the location bar,
- Navigate to brave://settings and verify that `Hide Brave Rewards button` toggle is present in the `Appearance` section,
- Toggle the setting and verify that the Brave Rewards action button is no longer shown in the location bar,
- Navigate to brave://rewards and click `Yes, I am in` button, wait for the rewards page to load,
- Verify that the Brave Rewards action button is now shown in the location bar,
- On the brave://rewards page, toggle `Brave Rewards` setting,
- Verify that the Brave Rewards action button is no longer shown in the location bar.

2. Test Guest/Tor profile
- Start Brave browser with a fresh profile and verify that the Brave Rewards action button is shown in the location bar,
- Click on the Avatar menu -> `Open Guest window`,
- Verify that the Guest window does not show the Brave Rewards action button,
- Close Guest window,
- In the original window, click on the Avatar menu -> `Open Tor window`,
- Verify that the Tor window does not show the Brave Rewards action button,
- Close Tor window.

## Reviewer Checklist:
